crates/biome_js_analyze/src/lint/nursery/use_consistent_boolean_props.rs

```diff
@@ -0,0 +1,223 @@
+            BooleanPropMode::Explicit => {
+                markup! { {prefix} "explicit ("<Emphasis>"={true}"</Emphasis>" must be present)." }
+            }
+        };
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            jsx_attribute.range(),
+            message.to_owned(),
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let jsx_attribute = ctx.query();
+        let mode = &ctx.options().mode;
+
+        let mut mutation = ctx.root().begin();
+        let mut mutation_has_changes = false;
+
+        match mode {
+            BooleanPropMode::Explicit => {
+                // Explicit mode: add `={true}` if missing
+                if jsx_attribute.initializer().is_none() {
+                    let JsxAttributeFields {
+                        name,
+                        initializer: _,
+                    } = jsx_attribute.as_fields();
+                    let name = name.ok()?;
+                    let name_syntax = name.syntax();
+
+                    // Move trailing trivia of name_syntax to close_curly_token
+                    let last_token_of_name_syntax = name_syntax.last_token()?;
+                    let next_last_token_of_name_syntax =
+                        last_token_of_name_syntax.with_trailing_trivia([]);
+                    let next_name = name.replace_token_discard_trivia(
+                        last_token_of_name_syntax,
+                        next_last_token_of_name_syntax,
+                    )?;
+
+                    let attr_value = make::jsx_expression_attribute_value(
+                        make::token(JsSyntaxKind::L_CURLY),
+                        biome_js_syntax::AnyJsExpression::AnyJsLiteralExpression(
+                            AnyJsLiteralExpression::JsBooleanLiteralExpression(
+                                make::js_boolean_literal_expression(make::token(T![true])),
+                            ),
+                        ),
+                        make::token(JsSyntaxKind::R_CURLY),
+                    );
+
+                    let next_attr = make::jsx_attribute(next_name)
+                        .with_initializer(make::jsx_attribute_initializer_clause(
+                            make::token(T![=]),
+                            AnyJsxAttributeValue::JsxExpressionAttributeValue(attr_value),
+                        ))
+                        .build();
+                    mutation_has_changes = true;
+                    mutation.replace_node(jsx_attribute.clone(), next_attr);
+                }
+            }
+            BooleanPropMode::Implicit => {
+                // Implicit mode: remove initializer if it is `true`
+                if let Some(init) = jsx_attribute.initializer()
+                    && let Ok(expr) = init.value()
+                    && let Some(value) = expr.as_jsx_expression_attribute_value()
+                    && value.to_trimmed_text() == "{true}"
```

Comment:
`to_trimmed_text` is able to allocate memory
also i don't think this would catch cases like `{     true  }`

Reply:
Plus, this check fails short if there are spaces or comments. So trivia. The check must change